internal/services/blueprints/blueprint_assignment_resource.go

```diff
@@ -181,54 +182,49 @@ func resourceBlueprintAssignmentCreateUpdate(d *pluginsdk.ResourceData, meta int
 				assignmentLockSettings.ExcludedActions = utils.ExpandStringSlice(excludedActionsRaw)
 			}
 		}
-		assignment.AssignmentProperties.Locks = assignmentLockSettings
+		a.Properties.Locks = assignmentLockSettings
 	}
 
-	identity, err := expandArmBlueprintAssignmentIdentity(d.Get("identity").([]interface{}))
+	i, err := identity.ExpandSystemOrUserAssignedMap(d.Get("identity").([]interface{}))
 	if err != nil {
 		return fmt.Errorf("expanding `identity`: %+v", err)
 	}
-	assignment.Identity = identity
+	a.Identity = *i
 
 	if paramValuesRaw := d.Get("parameter_values"); paramValuesRaw != "" {
-		assignment.Parameters = expandArmBlueprintAssignmentParameters(paramValuesRaw.(string))
+		a.Properties.Parameters = expandArmBlueprintAssignmentParameters(paramValuesRaw.(string))
 	} else {
-		assignment.Parameters = expandArmBlueprintAssignmentParameters("{}")
+		a.Properties.Parameters = expandArmBlueprintAssignmentParameters("{}")
 	}
 
 	if resourceGroupsRaw := d.Get("resource_groups"); resourceGroupsRaw != "" {
-		assignment.ResourceGroups = expandArmBlueprintAssignmentResourceGroups(resourceGroupsRaw.(string))
+		a.Properties.ResourceGroups = expandArmBlueprintAssignmentResourceGroups(resourceGroupsRaw.(string))
 	} else {
-		assignment.ResourceGroups = expandArmBlueprintAssignmentResourceGroups("{}")
+		a.Properties.ResourceGroups = expandArmBlueprintAssignmentResourceGroups("{}")
 	}
 
-	resp, err := client.CreateOrUpdate(ctx, targetScope, name, assignment)
-	if err != nil {
+	if _, err = client.CreateOrUpdate(ctx, id, a); err != nil {
 		return err
 	}
 
 	stateConf := &pluginsdk.StateChangeConf{
 		Pending: []string{
-			string(blueprint.Waiting),
-			string(blueprint.Validating),
-			string(blueprint.Creating),
-			string(blueprint.Deploying),
-			string(blueprint.Locking),
+			string(assignment.AssignmentProvisioningStateWaiting),
+			string(assignment.AssignmentProvisioningStateValidating),
+			string(assignment.AssignmentProvisioningStateCreating),
+			string(assignment.AssignmentProvisioningStateDeploying),
+			string(assignment.AssignmentProvisioningStateLocking),
 		},
-		Target:  []string{string(blueprint.Succeeded)},
-		Refresh: blueprintAssignmentCreateStateRefreshFunc(ctx, client, targetScope, name),
+		Target:  []string{string(assignment.AssignmentProvisioningStateSucceeded)},
+		Refresh: blueprintAssignmentCreateStateRefreshFunc(ctx, client, id),
 		Timeout: d.Timeout(pluginsdk.TimeoutCreate),
```

Comment:
we should be the remaining time from the context deadline, rather than the original timeout - since these can be different (e.g. with a timeout of 60m, if it takes 20m to create, we should be polling for 40m [using `time.Until(deadline)` rather than 60m)